Add the new AAD connection type and filtering feature to the symbol server docs (#2840)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/javascript/consume-source-maps-from-azure.md

Reviewer: global in article: 
Change "dropdown" to "dropdown list" or "dropdown menu"

per https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/d/dropdown

<!-- ====================================================================== -->
## Step 1: Enable source maps in DevTools

To make sure that source maps are enabled in DevTools:

1. To open DevTools, in Microsoft Edge, right-click a webpage, and then select **Inspect**.  Or, press **Ctrl+Shift+I** (Windows, Linux) or **Command+Option+I** (macOS).

1. In DevTools, click **Settings** (![Settings icon](./consume-source-maps-from-azure-images/settings-gear-icon-light-theme.png)) > **Preferences**.

1. Make sure the **Enable JavaScript source maps** checkbox and the **Enable CSS source maps** checkbox are selected:

    ![The Preferences configuration screen in DevTools settings](./consume-source-maps-from-azure-images/preferences-source-maps-devtools.png)

<!-- ====================================================================== -->

> ## Step 2 - Connect DevTools to the Azure Artifacts symbol server

Reviewer: ```suggestion
## Step 2: Connect DevTools to the Azure Artifacts symbol server
```
like Step 1.  Global in article: check headings "Step" punctuation.  Colon is more standard punctuation usage (though hyphen might be more "visible")

DevTools needs to be configured to successfully retrieve the published source maps for your website.

To configure DevTools, you create an Azure Artifacts symbol server connection. There are two types of connections: **Azure Active Directory**, which is the easiest to use, and **Personal Access Token**.

> #### Connect to the symbol server with Azure Active Directory

Reviewer: ```suggestion
#### Connect to the symbol server by using Azure Active Directory
```
for consideration.  
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/using-by-using-with - "Don't use _with_ as a synonym for _using_ or _by using_."

like the word "may", case-by-case judgment of tone/connotation.  I'd probably lean toward clarity: "by using"

1. In DevTools, click **Settings** (![Settings icon](./consume-source-maps-from-azure-images/settings-gear-icon-light-theme.png)) > **Symbol Server**.

1. Click **Add Symbol Server connection** to start creating a new connection.

> 1. In the **Authorization mode** dropdown, select **Azure Active Directory**.

Reviewer: ```suggestion
1. In the **Authorization mode** dropdown list, select **Azure Active Directory**.
```
Alts:
dropdown list
dropdown menu

per https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/d/dropdown

   If the **Azure Active Directory** option is disabled, sign in to Microsoft Edge. To learn more, see [Sign in to sync Microsoft Edge across devices](https://support.microsoft.com/microsoft-edge/sign-in-to-sync-microsoft-edge-across-devices-e6ffa79b-ed52-aa32-47e2-5d5597fe4674).

1. Under **Azure DevOps organization**, click **Select**.

   ![The Symbol Server setting screen in DevTools](./consume-source-maps-from-azure-images/select-aad-org.png)

   The list of DevOps organizations that you're a member of appears.

1. Double-click the organization that you want to connect to, or click it and then press **Enter**.

1. Click **Add** to create the connection.

   ![The Symbol Server setting screen in DevTools, showing the Add button to create a new connection](./consume-source-maps-from-azure-images/create-new-add-connection.png)

1. Click the **Close** (![Close settings icon](./consume-source-maps-from-azure-images/close-settings-icon.png)) button  in the upper right to close the **Settings** panel, and then click the **Reload DevTools** button.

> #### Connect to the symbol server with a Personal Access Token

Reviewer: ```suggestion
#### Connect to the symbol server by using a Personal Access Token
```

> To connect to the symbol server with a Personal Access Token (PAT), first generate a new PAT in Azure DevOps:

Reviewer: ```suggestion
To connect to the symbol server by using a Personal Access Token (PAT), first generate a new PAT in Azure DevOps:
```
consider per Comment https://github.com/MicrosoftDocs/edge-developer/pull/2840/files#r1330425973

To learn more about PAT, see [Use personal access tokens](/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate).

> With your new PAT generated, configure DevTools:

Reviewer: ```suggestion
Your new PAT has now been generated.  Next, configure DevTools, as follows:
```
could have a heading eg h4 here

1. In DevTools, click **Settings** (![Settings icon](./consume-source-maps-from-azure-images/settings-gear-icon-light-theme.png)) > **Symbol Server**.

1. Click **Add Symbol Server connection** to start creating a new connection.

1. In the **Authorization mode** dropdown, select **Personal Access Token**.

1. In the **Azure DevOps organization** text box, enter the Azure DevOps organization where you created the PAT.

1. In the **Personal access token** text box, paste your personal access token (PAT).

> 1. Click **Add** to create the connection.

Reviewer: ```suggestion
1. Click the **Add** button to create the connection:
```
* if it's a button, vs a link.
* colon if a lead-in for png.

> 1. Click **Add** to create the connection.
> 
>    ![The Symbol Server configuration screen in DevTools settings, showing the fields required to create a new PAT connection](./consume-source-maps-from-azure-images/create-new-pat-connection.png)

Reviewer: ```suggestion
   ![The Symbol Server configuration screen in DevTools settings, showing the fields required to create a new PAT connection](./consume-source-maps-from-azure-images/create-new-pat-connection.png)

1. Click the **Add** button. The connection is created.
```

1. Click the **Close** (![Close settings icon](./consume-source-maps-from-azure-images/close-settings-icon.png)) button  in the upper right to close the **Settings** panel, and then click the **Reload DevTools** button.

<!-- ====================================================================== -->
## Step 3: Retrieve original code in DevTools

> After the above setup steps, when you use DevTools to work on a build of your website for which symbols have been published, you can now see your original source code, instead of the transformed code.

Reviewer: ```suggestion
After doing the above setup steps, when you use DevTools to work on a build of your website for which symbols have been published, you can now see your original source code, instead of the transformed code.
```
* "doing" - more explicit

*  In the **Console** tool, links from log messages to source files go to the original files, not the compiled files.

*  When stepping through code in the **Sources** tool, the original files are listed in the **Navigator** pane on the left.

*  In the **Sources** tool, the links to source files that appear in the **Call Stack** of the **Debugger** pane open the original source files.

<!-- ====================================================================== -->
## Connect to multiple Azure Artifacts symbol servers

> You can configure DevTools to connect to multiple Azure Artifacts symbol servers if you need to fetch sourcemaps from different Azure DevOps organizations for your website.

Reviewer: ```suggestion
To fetch sourcemaps from multiple Azure DevOps organizations, you can configure DevTools to connect to multiple Azure Artifacts symbol servers.
```
alts:
To fetch sourcemaps from multiple Azure DevOps organizations for your website, you can configure DevTools to connect to multiple Azure Artifacts symbol servers.

1. In DevTools, click **Settings** (![Settings icon](./consume-source-maps-from-azure-images/settings-gear-icon-light-theme.png)) > **Symbol Server**.

1. Click **Add Symbol Server connection** to create an additional connection.

> 1. Refer to [Connect to the symbol server with Azure Active Directory](#connect-to-the-symbol-server-with-azure-active-directory) or [Connect to the symbol server with a Personal Access Token](#connect-to-the-symbol-server-with-a-personal-access-token) to configure the new connection.

Reviewer: ```suggestion
1. Configure the new connection.  See [Connect to the symbol server with Azure Active Directory](#connect-to-the-symbol-server-with-azure-active-directory) or [Connect to the symbol server with a Personal Access Token](#connect-to-the-symbol-server-with-a-personal-access-token).
```
Antipattern:
Do X, Y, and Z, to accomplish A.
Pattern (applies in some cases):
To accomplish A, do X, Y, and Z.

To edit or remove an existing connection, hover over the connection, and then click the **Edit** (![Edit icon](./consume-source-maps-from-azure-images/edit-connection-icon.png)) or **Remove** (![Remove icon](./consume-source-maps-from-azure-images/remove-connection-icon.png)) button next to the connection on the **Symbol Server** settings page.

![The remove and edit buttons, on a connection item, in the Symbol Server settings page](./consume-source-maps-from-azure-images/edit-remove-connection.png)

<!-- ====================================================================== -->
## Filter source maps by URL

You can decide which source maps to download, per Azure Artifacts symbol server connection, by using the **Filter behavior** feature.

1. In DevTools, click **Settings** (![Settings icon](./consume-source-maps-from-azure-images/settings-gear-icon-light-theme.png)) > **Symbol Server**.

1. Create a new connection or edit an existing connection.

1. In the **Filter behavior** dropdown, select either **Exclusion List** to exclude specific source maps, or **Inclusion List** to only include specific source maps.

> 1. In the text box below the dropdown, enter one URL per line for the source maps that you want to exclude or include.

Reviewer: ```suggestion
1. In the text box below the dropdown, enter one URL per line for the source maps that you want to exclude or include:
```
colon if this is a lead-in for png

![The remove and edit buttons, on a connection item, in the Symbol Server settings page](./consume-source-maps-from-azure-images/edit-remove-connection.png)

<!-- ====================================================================== -->
## Filter source maps by URL

To decide which source maps to download via the Azure Artifacts symbol server connection, use the **Filter behavior** feature.

1. In DevTools, click **Settings** (![Settings icon](./consume-source-maps-from-azure-images/settings-gear-icon-light-theme.png)) > **Symbol Server**.

1. Create a new connection or edit an existing connection.

1. In the **Filter behavior** dropdown list, select either **Exclusion List** to exclude specific source maps, or **Inclusion List** to only include specific source maps.

1. In the text box below the dropdown list, enter one URL per line for the source maps that you want to exclude or include:

   ![The inclusion list setting in a symbol server connection edit screen](./consume-source-maps-from-azure-images/filter-inclusion-list.png)

> 

Reviewer: ```suggestion

1. Click the **Save** button.

```

   The following wildcards are supported:

   | Wildcard | Meaning |
   |---|---|
   | ? | Matches a single character. |
   | * | Matches one or more of any character. |

> 

Reviewer: ```suggestion

1. Click the **Save** button.

```
